Account/FxAClient10.swift

```diff
@@ -398,14 +492,20 @@ extension FxAClient10: FxALoginClient {
         return makeRequest(mutableURLRequest) { FxAClient10.keysResponse(fromJSON: keyRequestKey, json: $0) }
     }
 
-    open func sign(_ sessionToken: Data, publicKey: PublicKey) -> Deferred<Maybe<FxASignResponse>> {
+    open func sign(_ sessionToken: Data, publicKey: PublicKey, isContentServer: Bool? = false) -> Deferred<Maybe<FxASignResponse>> {
         let parameters = [
             "publicKey": publicKey.jsonRepresentation() as NSDictionary,
             "duration": NSNumber(value: OneDayInMilliseconds), // The maximum the server will allow.
         ]
-
-        let URL = self.URL.appendingPathComponent("/certificate/sign")
-        var mutableURLRequest = URLRequest(url: URL)
+        
+        var url = self.authURL.appendingPathComponent("/certificate/sign")
+        
+        // Per FxA, to avoid having this request register another device record, we specify the service as
```

Comment:
Why would this request register another device record? Surely there has to be a better way to do this… can't we specify our existing device ID as part of the request?

Reply:
Look's like I was mistaken, it is only used for sessions that don't have an associated device, which this does. Ref: https://github.com/mozilla/fxa-content-server/issues/3933.

Reply:
OK, so what's the right thing to do here?

Reply:
Sorry should have mentioned I removed `isContentServer`.